src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Migrations/4.diff.sql

```diff
@@ -447,6 +487,130 @@ AS
+/***************************************************************************************/
+CREATE OR ALTER PROCEDURE dbo.GetExtendedQueryTagErrors (@tagPath VARCHAR(64))
+AS
+    SET NOCOUNT     ON
+    SET XACT_ABORT  ON
+    BEGIN TRANSACTION
+
+    DECLARE @tagKey		INT
+    DECLARE @watermark	BIGINT
+    
+    SELECT @tagKey = TagKey
+    FROM dbo.ExtendedQueryTag WITH(XLOCK)
+    WHERE dbo.ExtendedQueryTag.TagPath = @tagPath
+
+    -- Check existence
+    IF (@@ROWCOUNT = 0)
+        THROW 50404, 'extended query tag not found', 1 
+    ELSE
+        SELECT
+        XQTE.TagKey,
+        XQTE.ErrorCode,
+        XQTE.CreatedTime,
+        instance.StudyInstanceUid,
+        instance.SeriesInstanceUid,
+        instance.SopInstanceUid
+        FROM dbo.ExtendedQueryTagError AS XQTE
+        INNER JOIN dbo.Instance AS instance ON XQTE.Watermark = instance.Watermark
+        WHERE TagKey = @tagKey
+
+    COMMIT TRANSACTION
+GO
+
+/***************************************************************************************/
+-- STORED PROCEDURE
+--    DeleteExtendedQueryTagErrors
+--
+-- DESCRIPTION
+--    Delete errors for specified extended query tag
+--
+-- PARAMETERS
+--     @tagPath
+--         * The extended query tag path
+/***************************************************************************************/
+CREATE OR ALTER PROCEDURE dbo.DeleteExtendedQueryTagErrors (@tagPath VARCHAR(64))
+AS
+    SET NOCOUNT     ON
+    SET XACT_ABORT  ON
+    BEGIN TRANSACTION
+
+    DECLARE @tagKey INT
+
+    SELECT @tagKey = TagKey
+    FROM dbo.ExtendedQueryTag WITH(XLOCK)
+    WHERE dbo.ExtendedQueryTag.TagPath = @tagPath
+
+    -- Check existence
+    IF (@@ROWCOUNT = 0)
+        THROW 50404, 'extended query tag not found', 1 
+
+    DELETE FROM dbo.ExtendedQueryTagError
+    WHERE TagKey = @tagKey
+
+    COMMIT TRANSACTION
+GO
+
+/***************************************************************************************/
+-- STORED PROCEDURE
+--     AddExtendedQueryTagError
+--
+-- DESCRIPTION
+--    Add an Extended Query Tag Error
+--
+-- PARAMETERS
+--     @tagKey
+--         * The related extended query tag's key
+--     @createdTime
+--         * The time the error was created
+--     @errorCode
+--         * The error code
+--     @studyInstanceUid
+--         * study instance uid
+--     @seriesInstanceUid
+--         * series instance uid
+--     @sopInstanceUid
+--         * sop instance uid
+--     @sopInstanceKey
+--         * sop instance key
+/***************************************************************************************/
+
+CREATE OR ALTER PROCEDURE dbo.AddExtendedQueryTagError (
+    @tagKey INT,
+    @errorCode INT,
+    @Watermark BIGINT,
+    @createdTime DATETIME2(7)
+)
+AS
+    SET NOCOUNT     ON
+    SET XACT_ABORT  ON
+    BEGIN TRANSACTION
+
+     --Check if tag exists
+    IF NOT EXISTS (SELECT * FROM dbo.ExtendedQueryTag WHERE TagKey = @tagKey)
```

Comment:
We'll need a `HOLDLOCK` here and below to ensure the row isn't modified while we're adding its errors